rskj-core/src/main/java/co/rsk/logfilter/BlocksBloomProcessor.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.logfilter;
+
+import org.ethereum.core.Bloom;
+import org.ethereum.db.BlockStore;
+
+/**
+ * Created by ajlopez on 29/09/2020.
+ */
+public class BlocksBloomProcessor {
+    private final BlocksBloomStore blocksBloomStore;
+    private final BlockStore blockStore;
+
+    private BlocksBloom blocksBloomInProcess = null;
+
+    public BlocksBloomProcessor(BlocksBloomStore blocksBloomStore, BlockStore blockStore) {
+        this.blocksBloomStore = blocksBloomStore;
+        this.blockStore = blockStore;
+    }
+
+    public BlocksBloom getBlocksBloomInProcess() {
+        return this.blocksBloomInProcess;
+    }
+
+    public void processNewBlockNumber(long newBlockNumber) {
```

Comment:
Looks like this method is not thread-safe. Shouldn't it be `synchronized`?

Reply:
Currently only one thread at a time access this method, but yes, it could be fragile if new changes are added to the block processing logic (like processing blocks in multiple threads). So, done!